### PR TITLE
Initial version 

### PR DESCRIPTION
**Pull request by marcelosousa**
This PR introduces the files as they are specified in the initial version at: https://tour.golang.org/concurrency/10

**View on** [Reviewpad](https://beta.reviewpad.com/review/github.com/reviewpad/dejavus/pull/2)

### CODE REVIEW

File: fetcher.go [hunk 1]
```diff
@@ -0,0 +1,5 @@
+package main
+
+type Fetcher interface {
+	Fetch(url string) (body string, urls []string, err error)
```
Comment (reviewpad-team) [inline fetcher.go:4]: Could you provide some insight on this function?

Comment (marcelosousa) [reply fetcher.go:4]: Fetch returns the body of URL and a slice of URLs found on that page.

File: crawler.go [hunk 2]
```diff
@@ -0,0 +1,22 @@
+package main
+
+import "fmt"
+
+func Crawl(url string, depth int, fetcher Fetcher) {
```
Comment (marcelosousa) [inline crawler.go:5]: Crawl uses fetcher to recursively crawl pages starting with url, to a maximum of depth.

File: fakeFetcher.go [hunk 3]
```diff
@@ -0,0 +1,51 @@
+package main
+
+import "fmt"
+
+type fakeFetcher map[string]*fakeResult
```
Comment (marcelosousa) [inline fakeFetcher.go:5]: fakeFetcher is Fetcher that returns canned results.

File: fetcher.go (hunk 1 above)
Comment (marcelosousa) [inline fetcher.go:4]: Fetch returns the body of URL and a slice of URLs found on that page.

**Review by marcelosousa — COMMENTED** [review]
Comments to the main symbols.

**View on** [Reviewpad](https://beta.reviewpad.com/review/github.com/reviewpad/dejavus/pull/2#review-d11ab796-13e5-4735-8060-9b45499e0069)

**Pull request merged**